compiler/parser/src/string_parser.rs

```diff
@@ -36,6 +36,7 @@ impl<'a> StringParser<'a> {
         }
     }
 
+    #[inline]
     fn next_char(&mut self) -> Option<char> {
```

Comment:
Do we need this `inline` on `next_char`?

Reply:
its very likely this will get inlined either way (small-ish and private), annotating it doesn't hurt though.

Reply:
@DimitrisJim Got it, thanks!